### DISCUSSION TITLE
useWaitForTransaction always returns status: loading

### DISCUSSION THREAD

**Discussion by PheeWa** [wevm/wagmi#3706] [General]
I am facing an issue, the status return from useWaitForTransaction stays loading/isLoading: true forever even if the transaction is a success on Etherscan. It worked fine before yesterday. I also tried to get status by calling useTransaction hook, and the success status is returned correctly.

**Comment by OoXooOx** [comment]
try use your custom transport/RPC provider for fetch info from chain. Their native providers can work not properly.
import { w3mConnectors, w3mProvider } from '@web3modal/ethereum'
import { webSocket, createPublicClient } from "viem";
import { configureChains, createConfig } from 'wagmi'
import { goerli, mainnet, sepolia,  } from 'wagmi/chains';

export const walletConnectProjectId = ''

const { chains, 
  // publicClient,
   webSocketPublicClient } = configureChains(
  [goerli, sepolia], 
  [w3mProvider({ projectId: walletConnectProjectId })],
)

const transport = webSocket("wss: PASTE HERE YOUR PROVIDER FOR EXAMPLE FROM ALCHEMY without https" as string);
// const transport = webSocket("wss://eth-sepolia.g.alchemy.com/v2/AY..." as string);

const publicClient = createPublicClient({
  chain: sepolia,
  transport
});

export const config = createConfig({
  autoConnect: true,
  connectors: w3mConnectors({
    chains,
    projectId: walletConnectProjectId,
    version: 2,
  }),
  publicClient,
  webSocketPublicClient,
})

export { chains }
